core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -179,10 +177,9 @@ class Partition(val topicPartition: TopicPartition,
   // defined when this broker is leader for partition
   @volatile private var leaderEpochStartOffsetOpt: Option[Long] = None
   @volatile var leaderReplicaIdOpt: Option[Int] = None
-  @volatile var inSyncReplicas: Set[Int] = Set.empty[Int]
-  // Includes all valid broker ids (@see Request::isValidBrokerId) that contain logs. Doesn't contain
-  // future log replica id
-  @volatile var allReplicaIds: scala.collection.mutable.Set[Int] = scala.collection.mutable.Set(localBrokerId)
+  @volatile var inSyncReplicaIds = Set.empty[Int]
+  // Includes all valid broker ids (@see Request::isValidBrokerId) that were assigned to this topic partition.
+  @volatile var allReplicaIds = Seq.empty[Int]
```

Comment:
I'm guessing this the crux of the change? Maybe we should include a comment about order preservation here.

Reply:
Yes. I'll update the comment.

Reply:
Done.